launchable/utils/tracking.py

```diff
@@ -0,0 +1,98 @@
+from enum import Enum
+from typing import Dict, Any, Optional, Union
+
+from requests import Session
+from launchable.utils.http_client import _HttpClient, _join_paths
+
+from launchable.version import __version__
+
+
+class Tracking:
+    # General events
+    class Event(Enum):
+        SHALLOW_CLONE = 'shallow_clone'  # this event is an example
+
+    # Error events
+    class ExceptionEvent(Enum):
+        UNKNOWN_ERROR = 'UNKNOWN_ERROR'
+        INTERNAL_CLI_ERROR = 'INTERNAL_CLI_ERROR'
+        # Errors related to requests package
+        NETWORK_ERROR = 'NETWORK_ERROR'
+        TIMEOUT_ERROR = 'TIMEOUT_ERROR'
+        INTERNAL_ERROR = 'INTERNAL_ERROR'
+        UNEXPECTED_HTTP_STATUS_ERROR = 'UNEXPECTED_HTTP_STATUS_ERROR'
+
+    class Command(Enum):
+        VERIFY = 'VERIFY'
+        RECORD_TESTS = 'RECORD_TESTS'
+        RECORD_BUILD = 'RECORD_BUILD'
+        RECORD_SESSION = 'RECORD_SESSION'
+        SUBSET = 'SUBSET'
+
+
+class TrackingClient:
+    def __init__(self, command: Tracking.Command, base_url: str = "", session: Optional[Session] = None,
+                 test_runner: Optional[str] = "", dry_run: bool = False):
+        self.http_client = _HttpClient(
+            base_url=base_url,
+            session=session,
+            test_runner=test_runner,
+            dry_run=dry_run
+        )
+        self.command = command
+
+    def send_event(
+        self,
+        event_name: Union[Tracking.Event, Tracking.ExceptionEvent],
+        organization: str = "",
+        workspace: str = "",
+        metadata: Optional[Dict[str, Any]] = None
+    ):
+        if metadata is None:
+            metadata = {}
+        metadata["organization"] = organization
+        metadata["workspace"] = workspace
```

Comment:
Question: why do you set only organization and workspace?
Because you set other fields in send_error_event method, what is it different??

```
        metadata["stackTrace"] = stack_trace
        metadata["organization"] = organization
        metadata["workspace"] = workspace
        metadata["api"] = API
```

Reply:
Because I'd like to make it easier to set organization and workspace and api when calling `send_error_event` method.
We usually want to set them if possible. Thus, I designed it.

Reply:
However, if you feel strange, I'm ok with changing it.

Reply:
Ah, I see. send_event doesn't have api and stackTrace as receive params.
OK, you don't have to update it.